docs/developer/architecture/logs.rst

```diff
@@ -197,6 +197,444 @@ overview page.
+  ./charts/render.py fluentd --namespace metalk8s-logging charts/fluentd.yaml \
+    charts/fluentd/ > salt/metalk8s/addons/fluentd/deployed/charts.sls
+
+.. _fluentd helm chart: https://hub.helm.sh/charts/stable/fluentd
+
+Container images must be added in the buildchain
+``buildchain/buildchain/image.py`` and then replaced in
+``charts/loki-stack.yaml`` and ``charts/fluentd.yaml`` using the
+``__image__()`` macro to be replaced by jinja syntax when rendering the
+charts and then dynamically computed at run time.
+
+Fluent Bit
+""""""""""
+
+Fluent Bit will be deployed as a DaemonSet, because we need one collector on
+each control or workload plane node to be able to collect logs from both the
+Kubernetes platform and the applications that run on top of it.
+
+Loki
+""""
+
+Loki will be deployed as a StatefulSet as it needs persistent volumes to write
+the logs. It will be running only on infra nodes and we need at least
+2 replicas (except on a single node platform) of it to ensure its high
+availability.
+
+Loki HA
+_______
+
+Since we're using filesystem to store Loki's data, we have basically 2 ways
+for having multiple Loki instances running at the same time, with the same
+set of data.
+
+Either we do like Prometheus and we store everything on every instance of
+Loki, but it means we raise the storage, RAM and CPU needs for each additionnal
+instance, either we use a `new experimental feature`_ from Loki 1.5.0 where
+Loki ingesters use a hash ring and talk between them to route the queries to
+the right one. This approach needs an external KV store to work such as
+etcd or Consul.
+
+We choose to use the first approach as the second is not production ready
+and since we plan to only do short term retention on Loki, the impact on
+storage will not be that much important.
+Moreover, it eases the deployment and maintenance since there is not an
+extra component.
+
+.. _new experimental feature: https://github.com/grafana/loki/blob/master/docs/operations/storage/filesystem.md#new-and-very-experimental-in-150-horizontal-scaling-of-the-filesystem-store
+
+Fluentd
+"""""""
+
+Fluentd will be deployed as a Deployment with constraints to schedule it only
+on infra nodes.
+By default we need at least 2 replicas (except on a single node platform) to be
+sure we always have at least one instance of Fluentd running.
+
+Configuration
+~~~~~~~~~~~~~
+
+Fluent Bit
+""""""""""
+
+Fluent Bit needs to be configured to scrape and handle properly journal and
+containers logs by default.
+Its configuration
+
+For containers logs, we want to add the following labels:
+
+ - node: the node it comes from
+ - namespace: the namespace the pod is running in
+ - instance: the name of the pod
+ - container: the name of the container
+
+For journal we want these labels:
+
+ - node: the host it comes from
+ - unit: the name of the unit generating these logs
+
+.. note::
+
+  What if host != node ?
+
+  Do we want more labels (e.g. "app", "environment") ?
```

Comment:
since we have namespace and container, I would say it is sufficient for a first implem